src/modules/esl-media/control/esl-media-control-mixin.ts

```diff
@@ -0,0 +1,65 @@
+import {ESLMixinElement} from '../../esl-mixin-element/core';
+import {ESLTraversingQuery} from '../../esl-traversing-query/core/esl-traversing-query';
+import {listen, memoize} from '../../esl-utils/decorators';
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {evaluate} from '../../esl-utils/misc/format';
+
+import type {ESLMedia} from '../core/esl-media';
+
+export interface ESLMediaControlConfig {
+  target?: string;
+  action?: 'play' | 'pause' | 'stop' | 'toggle';
+  activeCls?: string;
+}
+
+@ExportNs('MediaControlMixin')
+export class ESLMediaControlMixin extends ESLMixinElement {
+  public static override is = 'esl-media-contol';
+
+  public static override observedAttributes = [ESLMediaControlMixin.is];
+
+  public static readonly DEFAULT_CONFIG: ESLMediaControlConfig = {
+    action: 'toggle'
+  };
+
+  @memoize()
+  public get config(): ESLMediaControlConfig {
+    const attrVal = this.$$attr(ESLMediaControlMixin.is) || '{}';
+    const userConfig = attrVal.trim().startsWith('{') ? evaluate(attrVal, {}) : {target: attrVal};
+    return {...ESLMediaControlMixin.DEFAULT_CONFIG, ...userConfig};
+  }
```

Comment:
I'd vote to have setter just in a simple way :
```ts
public set config(value: string | ESLMediaControlConfig) {
    const serialized = typeof value === 'string' ? value : JSON.stringify(value);
    this.$$attr(ESLMediaControlMixin.is, serialized); // It will trigger attribute change callback anyway
}
```